src/repositories/indexable-repository.php

```diff
@@ -93,8 +93,8 @@ public function __construct(
 		$this->term_builder              = $term_builder;
 		$this->home_page_builder         = $home_page_builder;
 		$this->post_type_archive_builder = $post_type_archive_builder;
-		$this->search_result_builder     = $search_result_builder;
-		$this->current_page_helper       = $current_page_helper;
+		$this->system_page_builder       = $system_page_builder;
+		$this->current_page              = $current_page_helper;
```

Comment:
Why remove the `_helper` part?

Reply:
Suggestion from @atimmer that I very much agree with. This means check on the current page are written as:

`$this->current_page->is_date_archive()` or `$this->current_page->is_404()` which reads a lot more smoothly.